src/monodroid/jni/xamarin_getifaddrs.cc

```diff
@@ -970,16 +986,16 @@ get_link_info (const struct nlmsghdr *message)
 
 	assert (message);
 	net_interface = reinterpret_cast <ifinfomsg*> (NLMSG_DATA (message));
-	length = message->nlmsg_len - NLMSG_LENGTH (sizeof (*net_interface));
+	length = static_cast<ssize_t>(message->nlmsg_len - NLMSG_LENGTH (sizeof (*net_interface)));
 	if (length <= 0) {
 		goto error;
 	}
-	
-	ifa = new _monodroid_ifaddrs (); //calloc (1, sizeof (*ifa));
```

Comment:
Ditto here.  After all the effort to replace `calloc` with `new`, why *undo* that change now?

Reply:
Ditto here :)